src/dotnet/ToolPackage/ToolPackageObtainer.cs

```diff
@@ -42,6 +41,79 @@ IProjectRestorer projectRestorer
             string targetframework = null,
             string source = null,
             string verbosity = null)
+        {
+            var stageDirectory = _toolsPath.WithSubDirectories(".stage", Path.GetRandomFileName());
+
+            var resourceManager = new TransactionResourceManager(
+                commit: () =>
+                {
+                    Directory.Move(
+                        stageDirectory.WithSubDirectories(packageId).Value,
+                        _toolsPath.WithSubDirectories(packageId).Value);
+
+                    Directory.Delete(stageDirectory.Value, true);
```

Comment:
This shouldn't be needed following a successful move, right?

Reply:
Oh never mind, we are restoring into a sub directory of this directory.  Is there a reason we can't just restore into the randomly-named stage directory itself?

Reply:
We did, move aasjdhal/my.tool/1.1 to .toolpkg/my.tool/1.1, and delete aasjdhal. We cannot just rename the random folder, there is more under .toolpkg

Reply:
Why restore to a sub-directory based on the package id at all if we're operating in a randomly named root directory in the first place?  I guess I'm not seeing why `.staged/<random>` -> `.toolspkg/<package-id>` wouldn't work.  It's no different from restoring into `.toolspkg/<package-id>` in the first place, right?

Reply:
what you suggested also works

Reply:
Changed to what you suggested